parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -334,81 +336,122 @@ object ParquetField {
-        case ParquetArray.default               => false
-        case ParquetArray.AvroCompat.avroCompat => true
-      }
-
-      override def buildSchema(cm: CaseMapper, properties: MagnolifyParquetProperties): Type = {
-        val repeatedSchema =
-          Schema.setRepetition(t.schema(cm, properties), Repetition.REPEATED)
-        if (isGroup(properties)) {
-          Types
-            .requiredGroup()
-            .addField(Schema.rename(repeatedSchema, AvroArrayField))
-            .as(LogicalTypeAnnotation.listType())
-            .named("iterable")
-        } else {
-          repeatedSchema
+      override def buildSchema(cm: CaseMapper, properties: MagnolifyParquetProperties): Type =
+        properties.writeArrayEncoding match {
+          case ArrayEncoding.Ungrouped =>
+            Schema.setRepetition(t.schema(cm, properties), Repetition.REPEATED)
+          case ArrayEncoding.TwoLevel =>
+            Types
+              .requiredGroup()
+              .addField(
+                Schema.rename(
+                  Schema.setRepetition(t.schema(cm, properties), Repetition.REPEATED),
+                  AvroArrayField
+                )
+              )
+              .as(LogicalTypeAnnotation.listType())
+              .named("iterable")
+          case ArrayEncoding.ThreeLevel =>
+            ConversionPatterns.listOfElements(
+              Repetition.REQUIRED,
+              t.schema(cm, properties).getName,
+              Schema.rename(t.schema(cm, properties), AvroElementField)
+            )
         }
-      }
 
       override protected def isGroup(properties: MagnolifyParquetProperties): Boolean =
-        avroCompatImported || properties.WriteAvroCompatibleArrays
+        properties.writeArrayEncoding != ArrayEncoding.Ungrouped
 
       override protected def isEmpty(v: C[T]): Boolean = v.forall(t.isEmpty)
 
       override def write(
         c: RecordConsumer,
         v: C[T]
-      )(cm: CaseMapper, properties: MagnolifyParquetProperties): Unit =
-        if (isGroup(properties)) {
-          c.startField(AvroArrayField, 0)
-          v.foreach(t.writeGroup(c, _)(cm, properties))
-          c.endField(AvroArrayField, 0)
-        } else {
-          v.foreach(t.writeGroup(c, _)(cm, properties))
+      )(cm: CaseMapper, properties: MagnolifyParquetProperties): Unit = {
+        properties.writeArrayEncoding match {
+          case ArrayEncoding.Ungrouped =>
+            v.foreach(t.writeGroup(c, _)(cm, properties))
+          case ArrayEncoding.TwoLevel =>
+            c.startField(AvroArrayField, 0)
+            v.foreach(t.writeGroup(c, _)(cm, properties))
+            c.endField(AvroArrayField, 0)
+          case ArrayEncoding.ThreeLevel =>
+            c.startField(AvroListField, 0)
+            c.startGroup()
+
+            v.foreach { elem =>
+              c.startField("element", 0)
+              t.write(c, elem)(cm, properties)
+              c.endField("element", 0)
```

Comment:
I see some warnings for test coverage for this new code block. Do we test anywhere the write path?

Reply:
good catch! that's on my to do list haha...